### ISSUE TITLE
W1101 false positive

### ISSUE THREAD

**Issue by tomasz-sodzawiczny** [aws-cloudformation/cfn-lint#4697]
### CloudFormation Lint Version

cfn-lint 1.56.1

### What operating system are you using?

Mac, Ubuntu

### Describe the bug

W1101 triggers on yaml that does not have an alias in it. Minimal repro attached - a snippet reduced from our real config.

Error message for search engines:
```
W1101 This code is using a YAML alias and can only be deployed using the 'package' cli command or AWS SAM
```

### Expected behavior

No warnings on yaml without aliases

### Reproduction template

```yaml
AWSTemplateFormatVersion: "2010-09-09"
Description: W1101 false positive

Transform: AWS::LanguageExtensions

Mappings:
  FooMap:
    TopKey:
      SecondKey: [0]

Resources:
  NoOp:
    Type: AWS::CloudFormation::WaitConditionHandle
    Metadata:
      A: !FindInMap [FooMap, TopKey, SecondKey]
      B: !FindInMap [FooMap, TopKey, SecondKey]
```
